File: challenge/java/Bin2Dec.java
import java.util.Scanner;

public class Bin2Dec {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.print("Enter binary number: ");

        String binary = input.next();

        int binaryLength = binary.length();

        if (binaryLength == 0 || binaryLength > 8) {
            System.out.println("Input must be 1-8 digits");
            input.close();
            return;
        }

        int decimal = 0;
        int exponent = binaryLength - 1;

        for (int i = 0; i < binaryLength; i++) {
            char charc = binary.charAt(i);

            if (charc != '0' && charc != '1') {
                System.out.println("Input can only be 0 and 1");
                input.close();
                return;
            }

            int digit = charc - '0';
            decimal += digit * (int) Math.pow(2, exponent);
            exponent--;
        }

        System.out.println("Decimal result: " + decimal);

        input.close();
    }
}

File: challenge/python/bin2dec.py
import sys

binary = input("Enter binary number: ")

binaryLength = int(len(binary))

if binaryLength == 0 or binaryLength > 8:
    print("Input must be 1-8 digits")
    sys.exit()

decimal = 0
exponent = binaryLength - 1

for char in binary:
    if char != '0' and char != '1':
        print("Input can only be 0 and 1")
        sys.exit()
    
    digit = int(char)
    decimal += digit * (2 ** exponent)
    exponent -= 1

print(f"Decimal result: {decimal}")